handlers/add_user_handler.go

```diff
@@ -43,30 +45,45 @@ func AddUserHandler(w http.ResponseWriter, r *http.Request, filePath string) {
 		return
 	}
 
-	// Read the existing users.yaml file
-	usersData, err := os.ReadFile(filePath)
-	if err != nil {
-		http.Error(w, "Failed to open users.yaml file", http.StatusInternalServerError)
+	requestToken := r.Header.Get("Authorization")
+	if requestToken == "" {
+		http.Error(w, "No authorization headers", http.StatusUnauthorized)
 		return
 	}
 
-	var users models.Users
-	err = yaml.Unmarshal(usersData, &users)
+	splitToken := strings.Split(requestToken, "Bearer ")
+	if len(splitToken) != 2 {
+		http.Error(w, "Invalid authorization header format", http.StatusUnauthorized)
+		return
+	}
+
+	requestToken = splitToken[1]
```

Comment:
For security reasons the token should also be encrypted. We need to encrypt using a string available in the Frontend and Backend. This string can be the env file, and it will be used to encrypt and decrypt strings. You can find an example of how to do this in this [gist](https://gist.github.com/manishtpatel/8222606)

Reply:
So I have to encrypt the token on top of using the GenerateSecureToken() and HashToken() functions in utils.go?